### PR TITLE
[4974] - 'My billing and shipping are the same' isn't selected - fixed

### PR DESCRIPTION
**Pull request by leonazava**
**Related issue(s):**
* Fixes scandipwa/scandipwa/issues/4974

**Problem:**
* Logic in isSameShippingAddress() method sometimes would return unexpected results, preventing the 'billing and shipping are same' box from being checked when it should've been.

**In this PR:**
* fixed 'billing and shipping are same' checkbox auto-selection (BASA) on billing page reload; 
* fixed BASA if the user chose a different SM from default that's different from billing; 
* fixed BAS not being selected when it should be

### CODE REVIEW

**Comment by leonazava** [comment]
request to QA: please test that [4716](https://github.com/scandipwa/scandipwa/pull/4716) and [4462](https://github.com/scandipwa/scandipwa/pull/4462) are still working. it was fine for me but better safe than sorry.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.container.js [hunk 1]
```diff
@@ -170,24 +192,33 @@ export class CheckoutBillingContainer extends PureComponent {
...
-            || (!default_billing)
-        )
-        && selectedShippingMethod !== STORE_IN_PICK_UP_METHOD_CODE;
+        if (selectedShippingMethod === STORE_IN_PICK_UP_METHOD_CODE) {
+            return false;
+        }
+
+        // if the user selected a shipping address different from default
+        if (newShippingId > 0) {
+            return (
```
[the remark this answers is not in the document]
Comment (leonazava) [reply packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.container.js]: done

**Review by zans-laksa — APPROVED** [review]

**Comment by Elena-Evsa** [comment]
Oops page on cart or checkout
image

**Comment by Elena-Evsa** [comment]
Fixed
4716 - not reproducible
4462 - not reproducible

**Pull request merged**